manim/utils/unit.py

```diff
@@ -0,0 +1,37 @@
+"""Implement the Unit class."""
+
+import numpy as np
+
+from .. import constants, config
+
+
+__all__ = ["Pixels", "Degrees", "Munits", "Percent"]
+
+
+class _PixelUnits:
+    def __mul__(self, val):
+        return val * config.frame_width / config.pixel_width
+
+    def __rmul__(self, val):
+        return val * config.frame_width / config.pixel_width
+
+
+class Percent:
+    def __init__(self, axis):
```

Comment:
Would it make sense to add another parameter, `mobject=None`, to `Percent` which would allow for measurements relative to other Mobjects, if one is passed?

I'm thinking that something like `50 * Percent(X_AXIS, my_square)` could be useful.

Reply:
+1 for this idea!

Reply:
:thinking: What would that mean exactly? 50% of the x coordinate of `my_square`?

Reply:
After some thought, I don't think we should include that in the `unit` module. The reason is that measuring relative distances or lengths is not really related to units. For example, if I want to say "move right by half of the distance between obj1 and obj2", then the only part that the unit module should be involved in is in the unit conversion, and not in computing what is the distance between obj1 and obj2.

Or did I misunderstand the suggestion?

Reply:
I understood it as it would be 50% of `my_square`'s width

Reply:
Oh I see :sweat_smile: I mean, that can be computed simply as `my_square.width / 2`. Again, no need to involve unit conversion when computing lengths.

Reply:
Okay, good point

Reply:
By the same logic, calculations involving `Percent` could also be handled just by using `config.frame_width` and `config.frame_height`, no?

My point was more about a uniform interface for all of these calculations.

Reply:
Ah that's a good point. In that case, by `50 * Percent(X_AXIS, my_square)` did you mean half the value of the X coordinate, or half the width?

Reply:
I would have understood it as the width.